### PR TITLE
Implement the payment method for tickets

### PR DESCRIPTION
**Pull request by ESteinman**
## PT-link:
https://www.pivotaltracker.com/story/show/160216893

## User story:
```
As a system owner
In order to increase the revenue flow
I would like to charge money for tickets
```

## Tasks
- Save `Ticket` into Database and display "shopping-cart" in `#header`
- Create a custom Stripe `payment-popup` to input card details. 
- message  with - payment is successful 

## New gems
gem `stripe-rails`
gem `acts_as_shopping_cart`
gem `stripe-ruby-mock`

## New Migrations
- CreateOrders
- CreateOrderItem
- AddUserToOrders
- AddStateToOrders

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,29 +1,41 @@
+# frozen_string_literal: true
+
```
Comment (faraznaeem) [inline app/controllers/application_controller.rb]: Remove comment :p

File: app/controllers/payments_controller.rb [hunk 2]
```diff
@@ -0,0 +1,57 @@
+# frozen_string_literal: true
+
```
Comment (faraznaeem) [inline app/controllers/payments_controller.rb]: :P

File: app/controllers/registrations_controller.rb [hunk 3]
```diff
@@ -6,15 +6,18 @@ def new
             render 'choice'
         else
             @role = params[:role] || 'fan'
-            super
+            build_resource
+            #yield resource if block_given?
+            #respond_with resource
+
```
Comment (faraznaeem) [inline app/controllers/registrations_controller.rb]: comments

File: app/models/order.rb [hunk 4]
```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
```
Comment (faraznaeem) [inline app/models/order.rb]: oh yes you know what to do

File: app/services/orders_service.rb [hunk 5]
```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+
```
Comment (faraznaeem) [inline app/services/orders_service.rb]: echo echo echo ....

File: app/views/purchase/create.html.haml [hunk 6]
```diff
@@ -0,0 +1,2 @@
+%h1 Purchase#create
+%p Find me in app/views/purchase/create.html.haml
```
Comment (faraznaeem) [inline app/views/purchase/create.html.haml]: Do you need this file?
If you do give it some other content

Comment (tochman) [reply app/views/purchase/create.html.haml]: @ESteinman ?? Wanna look into this?

File: app/services/orders_service.rb (hunk 5 above)
Comment (tochman) [reply app/services/orders_service.rb]: Rubocop does that to us....

File: features/step_definitions/basic_steps.rb [hunk 7]
```diff
@@ -25,13 +25,29 @@
...
+And('I fill in ticket fields') do
+  steps "
     And I fill in 'Fixed ticket price' with '100'
     And I fill in 'Ticket name' with 'Sure thing'
-  )
+  "
 end
 
-And ("I click on {string} in the confirmation popup") do |string|
+And ('I click on {string} in the confirmation popup') do |_string|
```
Comment (magnus-thor) [inline features/step_definitions/basic_steps.rb]: Unused variable

File: app/controllers/registrations_controller.rb (hunk 3 above)
Comment (tochman) [reply app/controllers/registrations_controller.rb]: @ESteinman Can you clean this up for me?

**Pull request merged**